### ISSUE TITLE
Parallel build failure with -j8 in src/amd/compiler/aco_instruction_selection_setup.cpp

### ISSUE THREAD

**Issue by akien-mga** [daniel-schuermann/mesa#35]
Building the current master branch (d12360227) on Mageia 7 x86_64, reusing the RPM spec file from Mageia's mesa package.

System:
CPU: Quad Core Intel Core i7-8705G (-MT MCP-) speed/min/max: 1503/800/4100 MHz Kernel: 5.1.14-desktop-1.mga7 x86_64 
Up: 13h 37m Mem: 4964.7/15792.2 MiB (31.4%) Storage: 953.87 GiB (43.7% used) Procs: 273 Shell: bash 4.4.23 inxi: 3.0.33

When I build with `ninja -v -j8 -C build`, I run into this error after a few hundred steps:
```
In file included from ../src/amd/compiler/aco_instruction_selection_setup.cpp:29:
../src/amd/common/sid.h:27:10: fatal error: amdgfxregs.h: No such file or directory
 #include "amdgfxregs.h"
          ^~~~~~~~~~~~~~
```

Full build log: [mesa-aco.build.log.txt](https://github.com/daniel-schuermann/mesa/files/3356553/mesa-aco.build.log.txt)

I managed to have it build *once* with -j8, and got it to fail quite reliably around 10 times. With -j2, I got it worked twice in a row.

It looks like `amdgfxregs.h` is not generated soon enough.

**Comment by pendingchaos** [comment]
Does this fix the issue:
```diff
diff --git a/src/amd/compiler/meson.build b/src/amd/compiler/meson.build
index cfb773d1feb..35f5f868573 100644
--- a/src/amd/compiler/meson.build
+++ b/src/amd/compiler/meson.build
@@ -89,7 +89,7 @@ libaco = static_library(
   ],
   dependencies : [
     dep_llvm, dep_thread, dep_elf, dep_libdrm_amdgpu, dep_valgrind,
-    idep_nir_headers,
+    idep_nir_headers, idep_amdgfxregs_h,
   ],
   c_args : [c_vis_args],
   cpp_args : [cpp_vis_args],
```
?

**Comment by akien-mga** [comment]
It seems to work indeed, I did 3 successful builds in a row with `-j8` (well not full builds, but ~700 steps while it used to fail at ~300, and `aco_instruction_selection_setup.cpp.o` is shown successfully compiled).

*Edit:* Did a complete build successfully too.

**Comment by daniel-schuermann** [comment]
Now, I know why I always have to build twice to get it completed o_O

**Comment by pendingchaos** [comment]
Should be fixed with 22b5ea1648b93ade5b5ae7f3af076393f1cb7455 (master) and 2a3b829800437521d6d0253d56ecb9c0a15d1a7b (backend).
